packages/extension/src/ui/features/accounts/accounts.state.ts

```diff
@@ -76,7 +76,9 @@ export const useAccountsOnNetwork = ({
 export const useAccount = (
   account?: BaseWalletAccount,
 ): Account | undefined => {
-  const accounts = useAccounts({ showHidden: true })
+  const network = useCurrentNetwork()
+  const allNetworks = account?.networkId !== network.id
+  const accounts = useAccounts({ showHidden: true, allNetworks })
```

Comment:
Nice catch 👍  Why this was not an issue before?

Reply:
This was an issue before when we are signing tx in a different network, probably more users use the wallet now so this corner case is tested now.